routing/index_graph.cpp

```diff
@@ -182,7 +153,21 @@ void IndexGraph::Import(vector<Joint> const & joints)
 void IndexGraph::SetRestrictions(RestrictionVec && restrictions)
 {
   ASSERT(is_sorted(restrictions.cbegin(), restrictions.cend()), ());
-  m_restrictions = move(restrictions);
+
+  m_restrictionsForward.clear();
+  m_restrictionsBackward.clear();
+
+  base::HighResTimer timer;
+  for (auto const & restriction : restrictions)
+  {
+    auto & forward = m_restrictionsForward[restriction.back()];
+    forward.emplace_back(restriction.begin(), std::prev(restriction.end()));
+    std::reverse(forward.back().begin(), forward.back().end());
+
+    m_restrictionsBackward[restriction.front()].emplace_back(std::next(restriction.begin()), restriction.end());
+  }
+
+  LOG(LDEBUG, ("Restrictions build done:", timer.ElapsedNano() / 1e6, "ms"));
```

Comment:
`LOG(LDEBUG, ("Restrictions are loaded in:", timer.ElapsedNano() / 1e6, "ms"));`?